pkg/anchor/handler/credential/handler.go

```diff
@@ -7,28 +7,129 @@ SPDX-License-Identifier: Apache-2.0
 package credential
 
 import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
 	"net/url"
 
 	"github.com/trustbloc/edge-core/pkg/log"
+	casapi "github.com/trustbloc/sidetree-core-go/pkg/api/cas"
+
+	"github.com/trustbloc/orb/pkg/store/cas"
 )
 
 var logger = log.New("anchor-credential-handler")
 
 // New creates new credential handler.
-func New(anchorCh chan []string) *AnchorCredentialHandler {
-	return &AnchorCredentialHandler{anchorCh: anchorCh}
+func New(anchorCh chan []string, casClient casapi.Client, httpClient *http.Client) *AnchorCredentialHandler {
+	return &AnchorCredentialHandler{anchorCh: anchorCh, casClient: casClient, httpClient: httpClient}
 }
 
 // AnchorCredentialHandler handles a new, published anchor credential.
 type AnchorCredentialHandler struct {
-	anchorCh chan []string
+	anchorCh   chan []string
+	casClient  casapi.Client
+	httpClient *http.Client
 }
 
 // HandleAnchorCredential handles anchor credential.
 func (h *AnchorCredentialHandler) HandleAnchorCredential(id *url.URL, cid string, anchorCred []byte) error {
-	logger.Debugf("received request id[%s], cid[%s], cred: %s", id, cid, string(anchorCred))
+	logger.Debugf("Received request: ID [%s], CID [%s], Anchor credential: %s", id, cid, string(anchorCred))
+
+	err := h.resolve(id, cid, anchorCred)
+	if err != nil {
+		return fmt.Errorf("failed to resolve anchor credential: %w", err)
+	}
 
 	h.anchorCh <- []string{cid}
 
 	return nil
 }
+
+// This will also save the credential in the local CAS.
+// TODO?: Will the Anchor Credential bytes we receive ever have differing amounts of whitespace? Will fields in the JSON
+//        come in a different order? If they do, their CIDs will be different, even though logically they're the same
+//        content.
+func (h *AnchorCredentialHandler) resolve(id *url.URL, cid string, anchorCred []byte) error {
```

Comment:
Will this be an issue?

Reply:
As discussed offline, it looks like from the point of view of the resolver, it doesn't need to worry about this.